wolfssl/wolfcrypt/port/atmel/atmel.h

```diff
@@ -108,21 +121,55 @@ int  atmel_get_rev_info(word32* revision);
 void atmel_show_rev_info(void);
 
 WOLFSSL_API int wolfCrypt_ATECC_SetConfig(ATCAIfaceCfg* cfg);
-
+#if defined(WOLFSSL_MICROCHIP_TA100)
+WOLFSSL_API int atmel_SetSharedDataConfig(ta_element_attributes_t* cfg);
+#endif
 /* The macro ATECC_GET_ENC_KEY can be set to override the default
    encryption key with your own at build-time */
 #ifndef ATECC_GET_ENC_KEY
     #define ATECC_GET_ENC_KEY(enckey, keysize) atmel_get_enc_key_default((enckey), (keysize))
 #endif
 int  atmel_get_enc_key_default(byte* enckey, word16 keysize);
 int  atmel_ecc_create_pms(int slotId, const uint8_t* peerKey, uint8_t* pms);
-int  atmel_ecc_create_key(int slotId, byte* peerKey);
+int  atmel_ecc_create_key(int slotId, int curve_id, byte* peerKey);
 int  atmel_ecc_sign(int slotId, const byte* message, byte* signature);
 int  atmel_ecc_verify(const byte* message, const byte* signature,
     const byte* pubkey, int* pVerified);
 
 #endif /* WOLFSSL_ATECC508A */
 
+#if defined(WOLFSSL_MICROCHIP_TA100)
+
+#if !defined(NO_AES) && defined(HAVE_AESGCM) && \
+    defined(WOLFSSL_MICROCHIP_AESGCM)
+#include <wolfssl/wolfcrypt/aes.h>
+
+int microchip_AesGcmEncrypt(Aes* aes, byte* out, const byte* in, word32 sz,
+                            const byte* iv, word32 ivSz,
+                            byte* authTag, word32 authTagSz,
+                            const byte* authIn, word32 authInSz);
+int microchip_AesGcmDecrypt(Aes* aes, byte* out, const byte* in, word32 sz,
+                            const byte* iv, word32 ivSz,
+                            const byte* authTag, word32 authTagSz,
+                            const byte* authIn, word32 authInSz);
+int microchip_aes_set_key(Aes* aes, const byte* key, word32 len,
+                          const byte* iv, int dir);
+void microchip_aes_free(Aes* aes);
+#endif /* !NO_AES && HAVE_AESGCM */
+#ifndef NO_RSA
+int microchip_rsa_create_key(RsaKey* key, int size, long e);
```

Comment:
Should all these be public functions? If not please add `WOLFSSL_LOCAL`.

Reply:
looks to be open still, we need to keep the boundary between what is a public API and local clear